### ISSUE TITLE
I hope can see the HLSL ver update

### ISSUE THREAD

**Issue by tkp206093** [bloc97/Anime4K#112]
**Is your feature request related to a problem? Please describe.**
I hope can see the HLSL ver update

**Describe the solution you'd like**
Because I very love use PotPlayer but PotPlayer not support GLSL ver

**Describe alternatives you've considered**
PotPlayer only can use HLSL (v0.9 Beta)

**Additional context**
HLSL ver update PLS!

**Comment by softbearlolz** [comment]
I hope this too

**Comment by TianZerL** [comment]
See #47

**Comment by Jules-A** [comment]
Yeah this is a dupe but I never got an answer on the other issue on whether the non-CNN shaders could be converted to HLSL.

**Comment by TianZerL** [comment]
#Jules-A
There is only one temporary texture can be used in HLSL of DirectShow based player, so it's not just CNN shaders that's hard to rewrite.

**Comment by Jules-A** [comment]
> 
> 
> @Jules-A
> There is only one temporary texture can be used in HLSL of DirectShow based player, so it's not just CNN shaders that's hard to rewrite.

Even for Dark/Thin Lines shaders? They are the ones I'm most interested combining with MadVR.

**Comment by bloc97** [comment]
Line thining already exists in MadVR, though it seems the implementation might be different. For line darkening it should be possible as it only needs 1 additional channel for storing data.

**Comment by Jules-A** [comment]
> Line thining already exists in MadVR, though it seems the implementation might be different. For line darkening it should be possible as it only needs 1 additional channel for storing data.

But it's not very good and doesn't allow tuning. For example, I run Anime4k's line thinning at 0.36 strength and 2 iterations, it's very different from the default settings although it does have it's strengths (720p+) and weakness (480p- or aliased).

**Comment by Durden-T** [comment]
me too
